app/uk/gov/hmrc/trackingconsentfrontend/assets/src/domain/userPreferenceFactory.ts

```diff
@@ -23,26 +26,36 @@ const setPreferences = preferencesIn => {
   })
 }
 
+const sanitizePreferences = preferences => {
+  const entries: [string, boolean][] = cookieTypes.reduce((accumulator, cookieType) => {
+    const { [cookieType]: value } = preferences
+
+    // If value is strictly not boolean, then do not assume either way
```

Comment:
Is there a way of making the code read clearly enough without a comment?  Would wrapping this in a well named function work?

Reply:
I wonder whether this would read well as a filter.  `.filter(booleanValues)`, `.filter(strictBooleanValues)` or `.filter(keepBooleanValues)`.